src/main/java/com/mate/fruitshop/strategy/ProcessSupplyTransaction.java

```diff
@@ -0,0 +1,25 @@
+package com.mate.fruitshop.strategy;
+
+import com.mate.fruitshop.dao.FruitStorageDao;
+import com.mate.fruitshop.dao.impl.FruitStorageDaoImpl;
+import com.mate.fruitshop.model.FruitEntry;
+import com.mate.fruitshop.model.Transaction;
+
+public class ProcessSupplyTransaction implements TransactionProcessingStrategy {
+    private final FruitStorageDao dao;
+
+    public ProcessSupplyTransaction() {
+        this.dao = new FruitStorageDaoImpl();
+    }
+
+    @Override
+    public boolean process(Transaction transaction) {
+        FruitEntry fruitEntry = dao.getEntryByFruitName(transaction.getFruitName());
+        if (fruitEntry == null) {
+            return dao.addFruitEntry(new FruitEntry(transaction.getFruitName(),

```

Comment:
Well, maybe supply can perform without fruits being on balance, but I'm not really sure.